network/random_test.go

```diff
@@ -0,0 +1,95 @@
+package network
+
+import (
+	"sync"
+	"testing"
+
+	gerr "github.com/gatewayd-io/gatewayd/errors"
+	"github.com/stretchr/testify/assert"
+)
+
+// TestNewRandom verifies that the NewRandom function properly initializes
+// a Random object with the provided server and its associated proxies.
+// The test ensures that the Random object is not nil and that the number of
+// proxies in the Random object matches the number of proxies in the Server.
+func TestNewRandom(t *testing.T) {
+	proxies := []IProxy{&MockProxy{}, &MockProxy{}}
+	server := &Server{Proxies: proxies}
+	random := NewRandom(server)
+
+	assert.NotNil(t, random)
+	assert.Equal(t, len(proxies), len(random.proxies))
+}
+
+// TestGetNextProxy checks the behavior of the NextProxy method in various
+// scenarios, including when proxies are available, when no proxies are available,
+// and the randomness of proxy selection.
+//
+// - The first sub-test confirms that NextProxy returns a valid proxy when available.
+// - The second sub-test ensures that an error is returned when there are no proxies available.
+// - The third sub-test checks if the proxy selection is random by comparing two subsequent calls.
+func TestGetNextProxy(t *testing.T) {
+	t.Run("Returns a proxy when proxies are available", func(t *testing.T) {
+		proxies := []IProxy{&MockProxy{}, &MockProxy{}}
+		server := &Server{Proxies: proxies}
+		random := NewRandom(server)
+
+		proxy, err := random.NextProxy()
+
+		assert.Nil(t, err)
+		assert.Contains(t, proxies, proxy)
+	})
+
+	t.Run("Returns error when no proxies are available", func(t *testing.T) {
+		server := &Server{Proxies: []IProxy{}}
+		random := NewRandom(server)
+
+		proxy, err := random.NextProxy()
+
+		assert.Nil(t, proxy)
```

Comment:
```suggestion
		assert.Nil(t, proxy)
		require.NotNil(t, err)
```

Otherwise next line will panic when there is no error

I'm unsure if the gerr implents Error()

If so, I would use


```suggestion
		assert.Nil(t, proxy)
		require.Error(t, err)
```